Sources/BothamViewController.swift

```diff
@@ -11,31 +11,34 @@ import Foundation
     import UIKit
 
 public class BothamViewController: UIViewController, BothamUI {
-    public var presenter: BothamPresenter! = nil
+    private var _presenter: BothamPresenter? = nil
 
     public override func viewDidLoad() {
         super.viewDidLoad()
-        presenter.viewDidLoad()
+        if self.respondsToSelector(Selector("presenter")) {
+            _presenter = self.performSelector(Selector("presenter")).takeUnretainedValue() as? BothamPresenter
+        }
+        _presenter?.viewDidLoad()
```

Comment:
Agree on the type thing, but what happen with force-casting of presenter as
properties.
We will call it twice.
Actually find by name is easier, but it shouldn't be complicated to
implement it either.

On Mon, Apr 11, 2016 at 4:59 PM, Pedro Vicente Gómez Sánchez <
notifications@github.com> wrote:

> In Sources/BothamViewController.swift
> <https://github.com/Karumi/BothamUI/pull/40#discussion_r59219506>:
>
> >
> >      public override func viewDidLoad() {
> >          super.viewDidLoad()
> > -        presenter.viewDidLoad()
> > +        if self.respondsToSelector(Selector("presenter")) {
> > +            _presenter = self.performSelector(Selector("presenter")).takeUnretainedValue() as? BothamPresenter
> > +        }
> > +        _presenter?.viewDidLoad()
>
> When you want to handle different part of the views with different
> presenters :) In addition, if I'm not wrong, starting all the presenters
> lifecycle methods is easier than to find the one with the name "presenter".
> I think we should use the type and not the name and the implementation
> based on the type and a list should be easier to implement.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Karumi/BothamUI/pull/40/files/b469cd2f2e6aeb0c19328dd1a57f079493a9a592#r59219506>
>